sos/plugins/openstack_ironic.py

```diff
@@ -95,6 +98,15 @@ class RedHatIronic(OpenStackIronic, RedHatPlugin):
         'openstack-ironic-discoverd-ramdisk'
     ]
 
+    inspector_package = 'openstack-ironic-inspector'
+
+    def add_introspection_data(self):
+        shell = (
+            "for uuid in $(openstack baremetal node list -f value -c UUID); "
+            "do openstack baremetal introspection data save $uuid; done"
+        )
```

Comment:
NACK: using shell syntax in `sos` plugins is absolutely forbidden (there's one historical example in the current tree - in the `rpm` plugin but new uses are completely banned).

This is just a bit lazy anyway - there is no reason at all that this cannot be implemented properly in Python, as other plugins (e.g. `networking`) do.

If for some reason this is some kind of "preferred form" of this data from upstream then a script should be provided that users or `sos` can run.

Reply:
Ok, I just wasn't sure if it's fine to run commands right from setup()

Reply:
Take a look at the `networking` plugin - it has a bunch of examples of running commands during `setup()` and then parsing the results. There are different interfaces to use depending on whether you want to also capture the data in the report, or if you just need it in-memory to figure out what other commands to run:

The `brctl` stuff uses `Plugin.get_cmd_output_now()`, which stores the result in the report and provides a path name string on return (absolute path into the `FileCacheArchive`):

https://github.com/sosreport/sos/blob/156e39ddf0126d1dee75087f3bfbd898db8f4692/sos/plugins/networking.py#L296

You can `open()` on it and do whatever you need to.

Alternately, the code to get ethernet interfaces does not store the result - it's held in memory only and thrown away. This uses the `call_ext_prog()` method:

https://github.com/sosreport/sos/blob/156e39ddf0126d1dee75087f3bfbd898db8f4692/sos/plugins/networking.py#L86

This family of calls return a Python dictionary object (conventionally stored in a variable named `something_result`). The keys are:

  * `status` numeric command status returned by wait/waitpid
  * `output` standard output, or combined stdout/stderr if `stderr=True`